src/Tizen.NUI/src/internal/WebView/WebContext.cs

```diff
@@ -690,5 +725,15 @@ private void OnPasswordDataListAcquired(IntPtr alist)
             }
             passwordDataListAcquiredCallback?.Invoke(passwordDataList);
         }
+
+        private void OnHttpRequestIntercepted(IntPtr interceptor)
+        {
+            if (httpRequestInterceptor != null)
+            {
+                httpRequestInterceptor.Dispose();
+            }
+            httpRequestInterceptor = new WebHttpRequestInterceptor(interceptor, true);
```

Comment:
Why do you store into instance variable? 
Why not local variable?
```suggestion
            var httpRequestInterceptor = new WebHttpRequestInterceptor(interceptor, true);
```

Reply:
If a local variable is used, a warning CA2000 occurs:

Warning	CA2000	Call System.IDisposable.Dispose on object created by 'new WebHttpRequestInterceptor(interceptor, true)' before all references to it are out of scope.

To fix this warning, a instance variable is stored.

Reply:
I think, interceptor object could be disposed when callback was completed
```suggestion
            using (var httpRequestInterceptor = new WebHttpRequestInterceptor(interceptor, true)) {
                      httpRequestInterceptedCallback?.Invoke(httpRequestInterceptor);
            }
```

Reply:
But WebHttpRequestInterceptor.WriteResponseChunk must be called outside WebContext.HttpRequestInterceptedCallback.

Reply:
Where is come from these constraint?

It is chromium-efl API description on ewk_intercept_request_response_write_chunk
I can't find this constraint
```
/**
 * @brief Writes a part of response body.
 *
 * @details This function can be used to write response body in chunks.
 *          Application doesn't have to prepare full response body before
 *          calling this function, it can be used as soon as a part of response
 *          is ready.
 *
 * @remarks If this function returns EINA_FALSE, handling the request is done.
 *          Any further calls on the Ewk_Intercept_Request instance result in
 *          undefined behavior. User should always check return value, because
 *          response to this request might not be needed anymore, and function
 *          can return EINA_FALSE even though user still has data to write.
 *
 *          It is only allowed to use this function *outside* of the
 *          #Ewk_Context_Intercept_Request_Callback.
 *
 *          If a part of response body has been written with this function, you
 *          can't use the following functions anymore:
 *          - ewk_intercept_request_response_set
 *          - ewk_intercept_request_response_body_set
 *
 *          Using this function with too big chunks of data defeats its
 *          purpose. Engine uses chunks of response written with this function
 *          without waiting for end of data, which is its main advantage over
 *          @a ewk_intercept_request_response_body_set.
 *
 * @since_tizen 3.0
 *
 * @param[in] intercept_request Intercept Request instance received from
 *            #Ewk_Context_Intercept_Request_Callback ewk_context callback
 * @param[in] chunk Part of response body for intercepted request
 * @param[in] length Length of response body part
 *
 * @return @c EINA_TRUE on success or EINA_FALSE on failure
 *
 * @pre Before writing response body, you should set response status
 *      and headers using the following functions:
 *      - ewk_intercept_request_response_status_set()
 *      - ewk_intercept_request_response_header_add()
 *      - ewk_intercept_request_response_header_map_add()
 *
 * @post After writing full response body in chunks using this function, call
 *       it again with NULL as @a chunk and 0 as @a length, to signal that
 *       response body is finished.
 *
 *       Use the same arguments if there is a failure with preparing the rest
 *       of response body on application side, and you don't want or can't
 *       continue writing parts.
 */
EXPORT_API Eina_Bool ewk_intercept_request_response_write_chunk(
    Ewk_Intercept_Request* intercept_request, const char* chunk,
    size_t length);

```

Reply:
In the comments above:
 *          It is only allowed to use this function *outside* of the
 *          #Ewk_Context_Intercept_Request_Callback.

Reply:
Oh, Thank you, I find it